tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -140,6 +142,47 @@ function get_container_runtime() {
 	fi
 }
 
+function is_containerd_capable_of_using_drop_in_files() {
+	local runtime="$1"
+
+	if [ "$runtime" == "crio" ]; then
+		echo "false" 
+		return
+	fi
+
+	if [[ "$runtime" =~ ^(k0s-worker|k0s-controller)$ ]]; then
+		# k0s does the work of using drop-in files better than any other "k8s distro", so
+		# we don't mess up with what's being correctly done.
+		echo "false"
+		return
+	fi
+
+	if [[ "$runtime" =~ ^(k3s|k3s-agent|rke2-agent|rke2-server)$ ]]; then
+		# k3s and rke2 are not behaving as expected with drop-in files yet.
+		# For now, let's skip using drop-in files for those, and revisit it
+		# after it works on their side
+		echo "false"
+		return
+	fi
+
+	local version=$(kubectl get node $NODE_NAME -o jsonpath='{.status.nodeInfo.containerRuntimeVersion}')
+	local version_major_minor=$(echo $version | grep -oE '[0-9]+\.[0-9]+')
+	local major=$(echo $version_major_minor | cut -d'.' -f1)
+	local minor=$(echo $version_major_minor | cut -d'.' -f2)
+
+	if [ $major -gt 1 ]; then
+		echo "true"
+		return
+	fi
+
+	if [ $major -eq 1 ] && [ $minor -ge 7 ]; then
+		echo "true"
+		return
+	fi
```

Comment:
I'm trying to decide if something like:
```
[ "1.7" != "$2" ] && [ "$(printf '%s\n' "$@" | sort -V | head -n 1)" == "1.7" ]
```
is better than this code (I think we have variants of this in different places. I think the answer is no though - it's definitely shorter, but that's not a good measure of code quality.

Reply:
It's shorter, indeed, but for my tired self looking at this code, I think the more clarity the better.